src/story_protocol_python_sdk/resources/IPAsset.py

```diff
@@ -89,13 +95,16 @@ def registerDerivative(self, child_ip_id, parent_ip_ids, license_terms_ids, lice
                 if not self.license_registry_client.hasIpAttachedLicenseTerms(parent_id, license_template, terms_id):
                     raise ValueError(f"License terms id {terms_id} must be attached to the parent ipId {parent_id} before registering derivative.")
 
+            # Fetch the current average gas price from the node plus 10%
+            current_gas_price = int(self.web3.eth.gas_price * 1.1)
```

Comment:
Allow user to provide specific gas price, if not use default (current gas price?)